pkgs/by-name/ll/lldap-cli/package.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+
+, curl
+, jq
+, makeWrapper
+, lldap
+, util-linux
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: rec {
+  name = "lldap-cli";
+
+  src = fetchFromGitHub {
+    owner = "ibizaman";
+    repo = finalAttrs.name;
+    rev = "422f67fbd9d52466591f7260592bbcd8fc09a210";
+    hash = "sha256-LX1g2OKrNI9XityQMW7Ag0O+rrFxWw5jczjDTkDLS+k=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    curl
+    jq
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/lldap-cli
+    mkdir -p $out/bin
+
+    cp lldap-cli $out/bin
+    chmod +x $out/bin/lldap-cli
+
+    substituteInPlace $out/bin/lldap-cli --replace "curl" "${curl}/bin/curl"
+    substituteInPlace $out/bin/lldap-cli --replace "jq" "${jq}/bin/jq"
+    substituteInPlace $out/bin/lldap-cli --replace "column" "${util-linux}/bin/column"
+    substituteInPlace $out/bin/lldap-cli --replace "lldap_set_password" "${lldap}/bin/lldap_set_password"
```

Comment:
There's probably a more elegant way of doing this. Maybe try replacing the shebang with a nix-shell with these dependencies?

Reply:
I took "inspiration" from the [manual](https://nixos.org/manual/nixpkgs/stable/#fun-makeWrapper). Specifically:

> # Be advised that paths often should be patched in directly
> # (via string replacements or in `configurePhase`).

That being said, I [just realized](https://nixos.org/manual/nixpkgs/stable/#fun-substitute) I can pass multiple `--replace` arguments to `substituteInPlace`. What do you think of:

```nix
substituteInPlace $out/bin/lldap-cli \
    --replace "curl" "${curl}/bin/curl" \
    --replace "jq" "${jq}/bin/jq" \
    --replace "column" "${util-linux}/bin/column" \
    --replace "lldap_set_password" "${lldap}/bin/lldap_set_password"
```

Reply:
The thing is that it's still pretty much doing the same thing and it's not exactly extensible. That said, I just realized that you could just add these to the nativeBuildInputs and the script should be able to use them just fine

Reply:
Happy to try something else but I don't have any ideas.

Reply:
Actually, try using something like https://github.com/nix-community/patsh or https://github.com/abathur/resholve. I wouldn't recommend using the latter anymore, though, since it depends on Python 2.7 which has been EOL for like 3-4 years now

Reply:
Used resholve. It actually found a bug :tada: https://github.com/ibizaman/lldap-cli/commit/65c4102ab0b314593d69c02a803a11ba1853978f